src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,259 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.io.FileWriter;
+import java.io.File;
+
+public class Duke {
+
+    public static final int MAX_TASK = 100;
+    public static final String COMMAND_LIST = "list";
+    public static final String COMMAND_DELETE = "delete";
+    public static final String COMMAND_MARK = "mark";
+    public static final String COMMAND_UNMARK = "unmark";
+    public static final String COMMAND_BYE = "bye";
+    public static final String COMMAND_EVENT = "event";
+    public static final String COMMAND_DEADLINE = "deadline";
+    public static final String COMMAND_TODO = "todo";
+    public static final File filePath = new File("src/main/java/duke/task/Hage.txt");
+
+
+    private static int itemNum;
+    private static String[] dateArray;
+    private static String[] eventArray;
+    private static final Scanner SCANNER = new Scanner(System.in);
+    private static ArrayList<Task> taskArrayList = new ArrayList<>();
+
+    public static void main(String[] args) throws DukeException, IOException {
+
+        printWelcomeMessage();
+        initTaskTable();
+
+        while (true) {
+            getNewInput();
+            saveTaskList();
+        }
+    }
+
+    private static void getNewInput() throws DukeException, IOException {
+        String inputCommand = SCANNER.nextLine();
+        executeCommand(inputCommand);
+    }
+
+    private static void executeCommand(String inputCommand) throws DukeException, IOException {
```

Comment:
This method is too long (>30 LoC). You could refactor the cases into separate methods and ensure [SLAP](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard).